### DISCUSSION TITLE
Chirality/Stereo info is lost in output 2D flat SDF

### DISCUSSION THREAD

**Discussion by RicardoGaGu** [rdkit/rdkit#5699] [Q&A]
RDkit version: 2022.03.5
Tested on: Ubuntu 20.04

I'm doing some standarization task on a bunch of molecules and my goal is to keep the input chirality when preparing the ligands. If the stereocenters are unspecified, do not assign anything, and if specified , preserve that information. The scenario is the following:

As input I have the following  2D flat SDF (No 3D coordinates) block:

 ```
 Mrv2120 10072201272D          

  5  4  0  0  0  0            999 V2000
   -4.9554    0.7738    0.0000 C   0  0  **1**  0  0  0  0  0  0  0  0  0
   -4.6455    0.0092    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0
   -4.1907    1.0836    0.0000 N   0  0  0  0  0  0  0  0  0  0  0  0
   -5.7200    0.4640    0.0000 S   0  0  0  0  0  0  0  0  0  0  0  0
   -5.2652    1.5384    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  1  **1**  0  0  0
  1  3  1  **6**  0  0  0
  1  4  1  0  0  0  0
  1  5  1  0  0  0  0
M  END
$$$$
```
Between double asterisks , you can see the atom parity flag in the V2000 SDF and the bond wedges as well. Indicating a specified chiral center.

The following code mimics the I/O operations. Somehow, RDkit internally perceives the chiral centers but does not write that information in the output 2D flat SDF:

```python
import rdkit
from rdkit import Chem
import os
import copy

input_file_path = "tests/chirality_block_test.sdf"
file_name = os.path.basename(input_file_path)
(name, ext) = os.path.splitext(file_name)
output_path= os.path.join("tests",name+"_out"+ext)
f_input = Chem.SDMolSupplier(input_file_path, removeHs = False)
f_output = Chem.SDWriter(output_path)

for mol in f_input:
    centers = Chem.FindMolChiralCenters(mol, force=True, includeUnassigned=True, useLegacyImplementation=False)
    print("Chiral centers: ",centers)
    std_mol = copy.deepcopy(mol)
    f_output.write(std_mol)`
```

The code only copies the mol object, but my goal is to apply a set of transformations (standarization), I just omitted these details because they do not affect.  I added a line to find the chiral centers, and to my surprise, RDkit seems to internally know chirality of the input molecule, but that information is not reflected in the output SDF:

`Chiral centers:  [(0, 'R')]`

The output 2D SDF does not contain the chirality tags, which is problem for interfacing with other compchem software:

```

     RDKit          2D

  5  4  0  0  0  0  0  0  0  0999 V2000
   -4.9554    0.7738    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   -4.6455    0.0092    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0
   -4.1907    1.0836    0.0000 N   0  0  0  0  0  0  0  0  0  0  0  0
   -5.7200    0.4640    0.0000 S   0  0  0  0  0  0  0  0  0  0  0  0
   -5.2652    1.5384    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  1  0
  1  3  1  0
  1  4  1  0
  1  5  1  1
M  END
$$$$
```

**Comment by greglandrum** [comment]
I don't believe that any information is being lost.
The output SDF that you show contains a wedged bond from atom 1 (the bond to atom 5), which is all that's required to indicate chirality on atom 1.
The RDKit does not write atom parity flags (the information in the parity flag is redundant with the information provided by 3D coordinates and/or bond wedges, so it's skipped), so there are no parity flags in the output.

**Reply by RicardoGaGu** [reply]
Ok, so I guess other software interfaces would infer the correct input chirality from the bond wedges. I did not see the equivalence. Thanks Greg! Would you recommend applying AssignStereochemistry(.) after doing some transformations (normalization, desalting, neutralizing).
